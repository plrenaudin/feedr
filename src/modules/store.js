import { observable, action, toJS, configure } from "mobx";
import db, { EMPTY_RECORD } from "./db";
import { shortFormatDate } from "./formatter";
import addDays from "date-fns/add_days";

configure({ enforceActions: true });

export default class Store {
  day = observable({ ...EMPTY_RECORD, date: shortFormatDate() });
  constructor() {
    this.load(shortFormatDate());
  }

  set = action(({ date = shortFormatDate(), morning = [], noon = [], evening = [] }) => {
    this.day.date = date;
    this.day.morning = morning;
    this.day.noon = noon;
    this.day.evening = evening;
  });

  addItem = action((meal, data) => {
    this.day[meal].push(data);
    this.save();
  });

  removeItem = action((meal, index) => {
    this.day[meal].splice(index, 1);
    this.save();
  });

  nextDay() {
    this.load(shortFormatDate(addDays(this.day.date, 1)));
  }

  previousDay() {
    this.load(shortFormatDate(addDays(this.day.date, -1)));
  }

  load = action(date => {
    return db.get(date).then(data => {
      this.set(data || { date });
    });
  });

  save() {
    return db.set(this.day.date, toJS(this.day));
  }
}
