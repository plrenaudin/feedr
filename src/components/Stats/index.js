import { inject } from "inferno-mobx";
import { Component } from "inferno";
import Pie from "../Pie";
import { DAY, MONTH, WEEK } from "../../modules/dateRangeEnum";

const defaultState = () => ({ data: [], selected: DAY });

class Stats extends Component {
  constructor(props) {
    super(props);
    this.store = props.store;
    this.state = defaultState();
  }

  fetchStats = range => () => {
    this.store.fetchStats(range).then(data =>
      this.setState({
        data,
        selected: range
      })
    );
  };

  componentDidMount() {
    this.fetchStats(DAY)();
  }

  componentWillUnmount() {
    this.setState(defaultState());
  }

  render() {
    return (
      <section class="stats">
        <ul class="control">
          <li class={this.state.selected === DAY && "selected"} onClick={this.fetchStats(DAY)}>
            Day
          </li>
          <li class={this.state.selected === WEEK && "selected"} onClick={this.fetchStats(WEEK)}>
            Week
          </li>
          <li class={this.state.selected === MONTH && "selected"} onClick={this.fetchStats(MONTH)}>
            Month
          </li>
        </ul>
        <Pie categories={this.state.data} />
      </section>
    );
  }
}

export default inject("store")(Stats);
