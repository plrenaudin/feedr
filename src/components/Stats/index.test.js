import { render } from "inferno";
import Stats from ".";
import store from "../../../test/fixture/store";
import { Provider } from "inferno-mobx";

it("Matches the snapshot", () => {
  const div = document.createElement("div");
  const rendered = render(
    <Provider store={store}>
      <Stats />
    </Provider>,
    div
  );
  expect(rendered.props.children.dom).toMatchSnapshot();
});
