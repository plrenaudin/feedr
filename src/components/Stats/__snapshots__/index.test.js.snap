// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot 1`] = `
<section
  class="stats"
>
  <ul
    class="control"
  >
    <li
      class="selected"
    >
      Day
    </li>
    <li
      class="false"
    >
      Week
    </li>
    <li
      class="false"
    >
      Month
    </li>
  </ul>
  <div
    class="pie"
  >
    <span
      class="icon-wrapper"
    >
      <svg
        class="icon"
      >
        <use
          xlink:href="#icon-question"
        />
      </svg>
    </span>
  </div>
</section>
`;
